### PR TITLE
[WIP] Mino onboarding styling

### PR DESCRIPTION
**Pull request by c-johnson**
Ongoing PR tracking development of UX & styling for the invite ticket flow.

### CODE REVIEW

File: src/bridge/components/Header.js [hunk 1]
```diff
@@ -40,15 +40,24 @@ const Crumbs = (props) => {
...
-    <Crumbs
-      routeCrumbs={ props.routeCrumbs }
-      skipRoute={ props.skipRoute }
-      networkType={ props.networkType }
-      wallet={ props.wallet }
-      pointCursor={ props.pointCursor }
-    />
-  </div>
+const Header = (props) => {
+  let showCrumbs = !props.routeCrumbs.contains(ROUTE_NAMES.INVITE_TICKET)
```
[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/components/Header.js]: Hmm. Why do you think so? I kinda like having the logic bundled with the component, so it's easy to find out how it's being rendered.

File: src/bridge/components/Passport.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  }
+
+  generateScreen() {
+    let blocks = [];
+
+    for (var i = 0; i < 80; i++) {
+      let rand = Math.round(Math.random())
+      let className = rand ? "passport-block bg-gray-30" : "passport-block"
+
+      blocks.push(<div key={i} className={className}></div>)
```
[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/components/Passport.js]: Huh. I definitely like this, but they don't seem to match the design exactly (not being perfect squares). Could try it out at next review meeting and see what people think.

File: src/bridge/lib/invite.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    });
+  })
+}
+
+export {
+  generateWallet,
+  downloadWallet,
+  INVITE_STAGES,
+  WALLET_STATES
+}
```
[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/lib/invite.js]: Yeah, large agree. I'll make sure to merge these at the end with existing functions that do similar things, there's quite a bit of overlap.

File: src/bridge/Bridge.js [hunk 4]
```diff
@@ -264,11 +264,20 @@ class Bridge extends React.Component {
 
     const View = router(routeCrumbs.peek())
 
+    let colClass
```
[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/Bridge.js]: Oh, absolutely no problem addressing these style comments! I figured I'd just do the laziest thing first and then clean up where people yelled at me :). Big fan of keeping styles consistent, happy to incorporate your comments here.

File: src/bridge/components/Passport.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Fang-) [reply src/bridge/components/Passport.js]: Yeah the half- and full-height ones seem a little funky. But the quadrant-based ones, (▞ and such) do render as perfect squares for me.

File: src/bridge/components/Header.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fang-) [reply src/bridge/components/Header.js]: It seems to me that a `showCrumbs=true/false` would be less opaque and easier to control than having that property decided by logic within the component. Like, if I want to make a new page that doesn't have breadcrumbs, then I don't want to jump around and modify component code. Does that make sense?

[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/components/Header.js]: Sure, it just seems rare you'd use the Header component outside of the one place it's in. But fair point, I'll pass it down.

File: src/bridge/views/InviteTicket.js [hunk 5]
```diff
@@ -144,8 +146,13 @@ class InviteTicket extends React.Component {
       let handler = isLogin ? this.handleInviteTicketInput : this.handleVerifyTicketInput
       let label = isLogin ? 'Activation code' : 'ticket'
 
+      const TicketElem =
```
Comment (jtobin) [inline src/bridge/views/InviteTicket.js]: Could consider this, which unconditionally avoids the naked `.value` access:

```
// if realWallet is Just, then VerifyTicketInput appropriate contents; if Nothing, remain Nothing
const Verified = this.state.realWallet.map(realWallet =>
  VerifyTicketInput(realWallet.ticket)
)

// Verified.getOrElse(TicketInput) falls back to TicketInput if Verified is Nothing
const TicketElem =
    isVerify
  ? Verified.getOrElse(TicketInput)
  : TicketInput
```

Comment (c-johnson) [reply src/bridge/views/InviteTicket.js]: Oh, very cool. Hadn't seen the `map` function before.

File: src/bridge/components/Passport.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/components/Passport.js]: Hmm. Still doesn't look like it fits right. This is using "full block" one; I'm not sure how you'd use the quadrant-based ones. Don't you need one block per bit?

image

[the remark this answers is not in the document]
Comment (Fang-) [reply src/bridge/components/Passport.js]: Can't you just like

```javascript
const blocks = ['▖','▗','▘','▙','▚','▛','▜','▝','▞','▟'];
for (length_of_line) {
  const r = newRandom(blocks.length);
  out += blocks[r];
}
```

File: src/bridge/views/InviteTicket.js [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+          autoFocus
+          name='invite'
+          placeholder={ `e.g. ${phTick}` }
+          value={ value }
+          onChange={ handler }>
+          <InnerLabel>{ label }</InnerLabel>
+        </TicketInput>
+      )
+    } else {
+      return null
```
[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/views/InviteTicket.js]: Huh. I'm fine with using Maybe values, but I admit don't have a clear sense for when variables should be wrapped. 
I feel like it's useful for values that get passed around a lot, like `web3` or `wallet`, where you want to do something other than return null when it gets unwrapped (like throw an error).

In this case, this function is just constructing DOM elements that get immediately consumed, where the proper way to unwrap it is to return null anyway.

Are there any general guidelines on when wrapping/unwrapping values is worth the overhead? I assume not every variable should be Maybe'd...

[the remark this answers is not in the document]
Comment (jtobin) [reply src/bridge/views/InviteTicket.js]: > In this case, this function is just constructing DOM elements that get immediately consumed, where the proper way to unwrap it is to return null anyway.

Derp, my bad.  I actually didn't notice this -- I think I had seen a few `nulls` here and there and just chose this one to comment on.  But this is a totally different case, agreed, and seems perfectly fine.

But yes, FWIW my own framework is just to never use `null`: for anything that could semantically be `null` in some circumstances, use `Maybe` instead; for a deliberate, exclusive, semantic `null`, use some other more type-appropriate "nully" construct instead (I'd probably use e.g. `<div />` in your example above, or `{}` if dealing with an empty object).

File: src/bridge/components/Passport.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/components/Passport.js]: Hmm. Gave it a shot. Doesn't quite match the mocks, but looks okay, except for these small subpixel seams between characters. We'll see what Jimmy thinks at the review.

image

**Comment by c-johnson** [comment]
Ok, PR comments have been addressed. Let me know if I've missed anything @jtobin @Fang-

File: src/bridge/components/Passport.js [hunk 7]
```diff
@@ -33,25 +33,25 @@ class Passport extends React.Component {
     let censor = "";
     const blocks = ['▖','▗','▘','▙','▚','▛','▜','▝','▞','▟'];
 
-    for (var i = 0; i < 12; i++) {
+    (new UInt32Array(12)).forEach((_, i) => {
```
Comment (Fang-) [inline src/bridge/components/Passport.js]: @jtobin is this what you like to see? ):

File: src/bridge/components/Passport.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (c-johnson) [reply src/bridge/components/Passport.js]: Jimmy prefers the original; gonna go ahead and add that back in. Sorry :p

File: src/bridge/views/InvitesManage.js [hunk 8]
```diff
@@ -1,24 +1,11 @@
...
-  EthereumWallet,
-  addressFromSecp256k1Public,
-  urbitWalletFromTicket,
-  addHexPrefix,
-  WALLET_NAMES
-} from '../lib/wallet'
-
-import ReactSVGComponents from '../components/ReactSVGComponents'
-import KeysAndMetadata from './Point/KeysAndMetadata'
-import Actions from './Point/Actions'
```
Comment (Fang-) [inline src/bridge/views/InvitesManage.js]: Thanks for cleaning up my messes. (^:

**Comment by c-johnson** [comment]
@Fang- Comments addressed! Found a couple of other issues that snuck in as well. 

One more thing to follow up on, did you ever figure out how to get the "Issued By" field from the invite wallet? Not sure if there's a specific azimuth call I should be making or if it's just in the wallet metadata somewhere..

**Comment by Fang-** [comment]
Yes, that should be [`azimuthjs.delegatedSending.invitedBy`](https://github.com/urbit/azimuth-js/blob/master/delegatedSending.js#L23-L28)!

**Comment by Fang-** [comment]
@c-johnson were you gonna still implement the `invitedBy` call, or is this good to go? Feel free to hit merge!

**Comment by c-johnson** [comment]
@Fang- I think the new flow forgoes this, so I'll just keep it out for now.

**Pull request merged**